### PR TITLE
AUT-12903-FR-CFN-resource-for-AWS-Stateful-Node

### PR DESCRIPTION
**Pull request by TamirYaariSpot**
_This is the default pull request template. You can customize it by adding a `pull_request_template.md` at the root of your repo or inside the `.github` folder._

# Jira Ticket

_Include a link to your Jira Ticket_
_Example: [AUT-12903](https://spotinst.atlassian.net/browse/AUT-12903)_

# Demo

_Please add a recording of the feature/bug fix in work. if you added new routes, the recording should show the request and response for each new/changed route_

# Checklist:
- [ ] I have filled relevant self assessment ([NodeJS](https://docs.google.com/forms/d/e/1FAIpQLSfl14u9AOBAmxVJ272tvO7XNuXE-EMvWaGcaRZalu1UAKB7RA/viewform), [Frontend](https://docs.google.com/forms/d/e/1FAIpQLSdiBPNKH81w_EkavihVL8Uwb0j7tP8PwJmLFYm2nCOQxz-1qw/viewform), [Backend](https://docs.google.com/forms/d/e/1FAIpQLSed_PsTJ5-XIWkFL6BSDE2AQRBVPmwc3PAmHZkUn-erzVI37Q/viewform?usp=sf_link))
- [ ] I have run ESlint on my changes and fixed all warnings and errors (**NodeJS & Frontend Services**)
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have validated all the requirements in the Jira task were answered
- [ ] I have all neccessary approvals for the design/mini design of this task
- [ ] I have approved the API changes and granular permission patterns (documentation subtask) (**For public services only**) 

[AUT-12903]: https://spotinst.atlassian.net/browse/AUT-12903?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/util.js [hunk 1]
```diff
@@ -1050,6 +1062,111 @@ module.exports.rollCluster = function (obj) {
...
+
+  request(getStatefulNodeOptions, function (err, res, body) {
+    module.exports.validateResponse({
+      err:       err,
+      res:       res,
+      body:      body,
+      event:     event,
+      context:   context,
+      successCb: function (spotResponse) {
+        console.log("stateful node does exist, fail the delete.");
```
Comment (Guybenisrael) [inline lib/util.js:1101]: success logs should be different

File: lib/util.js [hunk 2]
```diff
@@ -1050,6 +1062,111 @@ module.exports.rollCluster = function (obj) {
...
+      'User-Agent':    module.exports.getUserAgent()
+    }
+  };
+
+  request(getStatefulNodeOptions, function (err, res, body) {
+    module.exports.validateResponse({
+      err:       err,
+      res:       res,
+      body:      body,
+      event:     event,
```
Comment (Guybenisrael) [inline lib/util.js:1098]: no resource field?

File: lib/util.js [hunk 3]
```diff
@@ -1050,6 +1062,111 @@ module.exports.rollCluster = function (obj) {
...
+ * @name validateOceanCluster
+ *
+ * @param {String} statefulNodeId - Stateful Node ID
+ * @param {String} token - Spotinst API Token
+ * @param {Object} event - Event data from Lambda
+ * @param {Object} context - Context data from Lambda
+ *
+ * @property {String} asg name string
+ */
+module.exports.validateStatefulNode = function (statefulNodeId, token, event, context) {
```
Comment (Guybenisrael) [inline lib/util.js:1079]: add resume and pause

File: lib/resources/statefulNode/update.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        request(updateStatefulNodeOptions, function (err, res, body) {
+            spotUtil.validateResponse({
+                err: err,
+                res: res,
+                body: body,
+                event: event,
+                context: context,
+                resource: 'statefulNode',
+                action: 'update',
+                successCb: function () {
```
Comment (Guybenisrael) [inline lib/resources/statefulNode/update.js:77]: add failuredb

File: lib/util.js (hunk 2 above)
Comment (TamirYaariSpot) [reply lib/util.js:1098]: done

**Pull request merged**